src/dev/generate_release_note_helper.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { resolve } from 'path';
+
+export const filePath = resolve(__dirname, '..', '..', 'CHANGELOG.md');
+export const fragmentDirPath = resolve(__dirname, '..', '..', 'changelogs', 'fragments');
+export const fragmentTempDirPath = resolve(__dirname, '..', '..', 'changelogs', 'temp_fragments');
+export const releaseNotesDirPath = resolve(__dirname, '..', '..', 'release-notes');
+
+export function getCurrentDateFormatted(): string {
+  const now = new Date();
+  const year = now.getFullYear();
+  const month = now.getMonth() + 1;
+  const day = now.getDate();
+
+  const formattedMonth = month.toString().padStart(2, '0');
+  const formattedDay = day.toString().padStart(2, '0');
+
+  return `${year}-${formattedMonth}-${formattedDay}`;
+}
+
+export const SECTION_MAPPING = {
+  breaking: '💥 Breaking Changes',
+  deprecate: 'Deprecations',
+  security: '🛡 Security',
+  feat: '📈 Features/Enhancements',
+  fix: '🐛 Bug Fixes',
+  infra: '🚞 Infrastructure',
+  doc: '📝 Documentation',
+  chore: '🛠 Maintenance',
+  refactor: '🪛 Refactoring',
+  test: '🔩 Tests',
+};
+
+export type SectionKey = keyof typeof SECTION_MAPPING;
+export type Changelog = Record<SectionKey, string[]>;
+
+const MAX_ENTRY_LENGTH = 100;
+
+// validate format of fragment files
+export function validateFragment(content: string) {
+  const sections = content.split('\n\n');
```

Comment:
These need to accommodate `\r\n\r\n` combos as well. 

Maybe
```suggestion
  const sections = content.split(/(?:\r?\n){2,}/);
```

Reply:
@CMDWillYang can you check this?

Reply:
fixed, thanks!